docs/samples-and-tutorials/index.md

```diff
@@ -1,111 +1,122 @@
-[ASP.NET site](https://docs.microsoft.com/aspnet/core/tutorials/).
+Many topics show source code and samples that are available for viewing or download from GitHub. To view a sample, follow the sample link to the sample folder at aspnet/Docs or dotnet/docs on GitHub. To obtain the code for a sample mentioned in a topic, follow these instructions:
 
-In addition, if you are interested in exploring C# using an
-online playground, try these [interactive tutorials](http://go.microsoft.com/fwlink/?LinkId=817234).
+1. Obtain the repository containing the sample code for your local system by performing one of the following procedures:
+   * [Fork](https://help.github.com/articles/fork-a-repo/) the repository and [clone](https://help.github.com/articles/cloning-a-repository/) the fork to your local system. Forking and cloning permits you to make contributions to the .NET documentation by committing changes to your fork and then creating a pull request for the official docs repository. For more information, see the [.NET Documentation Contributing Guide](https://github.com/dotnet/docs/blob/master/CONTRIBUTING.md) and the [ASP.NET Docs Contributing Guide](https://github.com/aspnet/Docs/blob/master/CONTRIBUTING.md).
+   * Clone the repository locally. Note that if you clone a docs repository directly to your local system that you won't be able to make commits directly against the official repository, so you won't be able to make documentation contributions later. Use the fork and clone procedure previously described if you want to preserve the opportunity to contribute to the documentation later.
+   * Download a ZIP of the repository to your local system. Un-ZIP the compressed archive.
+1. Navigate within the repository to the sample's location. The relative path to the sample's location appears in the GitHub UI and in your browser's address bar when you follow the link to the sample in the topic where the sample is described.
+1. If you intend to run the sample code locally and prefer to run it from a different location on your system than where you maintain your repository files, copy the sample's files and folders to your preferred location.
+1. To run a sample, you have several options:
+   * Use the [dotnet CLI tools](../core/tools/index.md): In a console window, navigate to the sample's folder and use dotnet CLI commands.
+   * Use [Visual Studio](https://www.visualstudio.com/) or [Visual Studio for Mac](https://www.visualstudio.com/vs/visual-studio-mac/): Open the sample by selecting **File > Open > Project/Solution** from the menu bar, navigate to the sample project folder, and select the project file (*.csproj* or *.fsproj*).
+   * Use [Visual Studio Code](https://code.visualstudio.com/): Open the sample by selecting **File > Open Folder** from the menu bar and selecting the sample's project folder.
+   * Use a different IDE that supports .NET Core projects.
 
-## General
+## .NET Core
 
 ### Samples
```

Comment:
> I'd consider putting the how-to-download section at the end and then link to that anchor from sample links in docs. It seems obtrusive to put it up front when it isn't really the primary focus of the article.

I originally had it at the end. I moved it up because you probably need to be aware of this information *before* you start clicking on the sample repos. Otherwise, there could (a) be a single sentence in the opening paragraph that says "for info on downloading, see section" and link it to the section at the bottom, or (b) every one of these sample README's need a blurb about how to get the code downloaded and that would be linked back to that section at the bottom of this doc.